etc/cmake/options.cmake

```diff
@@ -132,7 +132,7 @@ endif()
 
 option(OT_EXTERNAL_HEAP "enable external heap support")
 if(OT_EXTERNAL_HEAP)
-    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_EXTERNAL_HEAP_ENABLE=1")
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_HEAP_EXTERNAL_ENABLE=1")
```

Comment:
Should also change the cmake option to `OT_HEAP_EXTERNAL`?

Reply:
That would have an impact of users of the config API, e.g. Zephyr. So it would need some sort of deprecation warning.